### ISSUE TITLE
Improve docs for --files-from option

### ISSUE THREAD

**Issue by BCITMike** [ncw/rclone#547]
When filing an issue, please include the following information if
possible as well as a description of the problem.

> What is your rclone version (eg output from `rclone -V`)
rclone v1.30
> Which OS you are using and how many bits (eg Windows 7, 64 bit)
rasperry pi debian jessie lite
> Which cloud storage system are you using? (eg Google Drive)
Google Drive
> The command you were trying to run (eg `rclone copy /tmp remote:tmp`)
sudo rclone --files-from somefile.txt copy gdrive:/pi/
sudo rclone copy --files-from somefile.txt copy gdrive:/pi/

> A log from the command with the `-v` flag (eg output from `rclone -v copy /tmp remote:tmp`)

```
It is only necessary to use a unique prefix of the subcommand, eg 'mo'
for 'move'.
Command copy needs 2 arguments mininum

```

I can't figure out the syntax to use the --files-from option.  The documentation mentions:

> --files-from - Read list of source-file names
> This reads a list of file names from the file passed in and only these files are transferred. The filtering rules are ignored completely if you use this option.
> 
> Prepare a file like this files-from.txt
> 
> file1.jpg
> file2.jpg
> Then use as --files-from files-from.txt. This will only transfer file1.jpg and file2.jpg providing they exist.

However, I can't get that to work. What is the correct syntax to use?

This works:
sudo rclone copy /etc/ethers gdrive:/pi/
sudo rclone copy /etc/hosts gdrive:/pi/
sudo rclone copy /etc/pihole/whitelist.txt gdrive:/pi/
sudo rclone copy /etc/pihole/blacklist.txt gdrive:/pi/

I'm just expecting to be able to put those 4 source paths into a single file, and then call that single file.  What am I missing?  I've tried without paths and simple filename inside somefile.txt, no difference.

Thanks

**Comment by BCITMike** [comment]
Oops, should have spent a bit more time before opening an issue.  Looks like I need to break this up into two commands if I want them all in one folder, or use /etc/ as the copy source dir to only get a pihole subdir on the destination.

files-from.txt contents:
ethers
hosts
pihole/whitelist.txt
pihole/blacklist.txt

command:
sudo rclone --files-from files-from2.txt copy /etc gdrive:/pi/

result on G Drive:
\pi\ethers
\pi\hosts
\pi\pihole\blacklist.txt
\pi\pihole\whitelist.txt

**Comment by ncw** [comment]
Looking at the docs for `--files-from` I can see that they could do with improving.

I'll use this issue to do that :-)

**Comment by CottonEaster** [comment]
&nbsp;&nbsp;&nbsp;&nbsp;I feel that that the `--filter-from` documentation could be improved too. For one thing, the example file given there contains the text '# a sample exclude rule file', which seems wrong. For another thing, it is not clear whether or not `--filter-from` files can include _folders_ or only files. Further: how come `--filter-from` files use `*` for exclusions and not, as rclone does it elsewhere, `**`?

&nbsp;&nbsp;&nbsp;&nbsp;I find the documentation very confusing and consequently I am unable to tell whether my backup scripts are faulty or whether rclone is buggy. I can see you've put work into the documentation but, for a backup program, I do think the documentation needs to be clearer.

**Comment by ncw** [comment]
@CottonEaster wrote
>  For one thing, the example file given there contains the text '# a sample exclude rule file', which seems wrong.

I've fixed that now - thanks!

> For another thing, it is not clear whether or not --filter-from files can include folders or only files. Further: how come --filter-from files use * for exclusions and not, as rclone does it elsewhere, **

The filter-from file contains filter rules - they aren't special in any way.  so you can include directories, use `*` or `**` as appropriate.  I put some more examples in the file to try to help.

> I find the documentation very confusing and consequently I am unable to tell whether my backup scripts are faulty or whether rclone is buggy. I can see you've put work into the documentation but, for a backup program, I do think the documentation needs to be clearer.

Filtering is complicated and powerful.  I'm always glad to receive feedback about the docs - especially updates or re-wording to make them clearer.

I put my changes here 58f7b4ed7c17b3f8b317e33bafa5d1e9bc683d38

PS is your handle to do with the [cotoneaster plant](https://en.wikipedia.org/wiki/Cotoneaster)?  I thought I was the only person who called it a Cotton Easter (to pull the leg of my botanist wife!).

**Comment by CottonEaster** [comment]
> PS is your handle to do with the cotoneaster plant? I thought I was the only person who called it a Cotton Easter (to pull the leg of my botanist wife!).

Not really. What we have here is an extraordinary co-incidence. To explain: I thought up the name pretty much randomly, and then searched the web for it to check I had not chosen a name that did actually mean anything or at least not anything I did not wish to evoke. At that point I discovered the plant. And here you are with that connection to the plant.
